### PR TITLE
Add conda

### PR DESCRIPTION
**Pull request by radujica**
* add conda recipe for publishing
* add conda environment for dev setup and instructions
* clean/separate requirements

Looks like we can't do pure conda for now :(
https://github.com/alkaline-ml/pmdarima/issues/326

For publishing/CD (both pypi and conda) we could depend on github releases and setup github action, as long as we sync setup.py version with a git tag (e.g. bump2version?). Actual release creation can trigger it, so it's still manual, but no individual developer setup would be required.

Other notes:
`conda-build conda_recipe --python 3.x`
`conda convert --platform all <path-to-bz2>`
`conda-verify <path-to-bz2>`

### CODE REVIEW

File: conda_recipe/u8timeseries/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+build:
+  number: 0
+  script: '{{ PYTHON }} -m pip install . --no-deps -vv'
+
+requirements:
+  host:
+    - python >=3.6
+  run:
+  # only essential requirements to use this library ~ setup.py install_requires ~ requirements/main.txt
```
Comment (hrzn) [inline conda_recipe/u8timeseries/meta.yaml:19]: Isn't there a way to tie this to `requirements/main.txt` ?

File: .github/workflows/workflow.yml [hunk 2]
```diff
@@ -22,13 +22,14 @@ jobs:
         id: cache
         with:
           path: ~/.cache/pip
-          key: ${{ runner.os }}-pip-${{ hashFiles('requirements/main.txt') }}
+          key: ${{ runner.os }}-pip-${{ hashFiles('requirements/main.txt', 'requirements/tests.txt') }}
```
Comment (TheMP) [inline .github/workflows/workflow.yml:25]: Pls align with https://github.com/unit8co/u8timeseries/pull/49

File: requirements/publish.txt [hunk 3]
```diff
@@ -0,0 +1 @@
+twine
```
Comment (TheMP) [inline requirements/publish.txt:1]: pls freeze the version

File: conda_recipe/u8timeseries/meta.yaml (hunk 1 above)
Comment (radujica) [reply conda_recipe/u8timeseries/meta.yaml]: Made it work :)

I can now do jinja 'coding', lolz

File: requirements/tests.txt [hunk 4]
```diff
@@ -0,0 +1 @@
+testfixtures
```
Comment (TheMP) [inline requirements/tests.txt:1]: I feel safer when the versions are frozen :)

File: setup.py [hunk 5]
```diff
@@ -1,24 +1,23 @@
...
+      version=open('u8timeseries/VERSION', 'r').read(),
       description='A collection of easy-to-use timeseries forecasting models',
       url='http://github.com/unit8co/u8timeseries',
       author='Unit8 SA',
       author_email='info@unit8.co',
       license='Apache License 2.0',
       packages=find_packages(),
-      install_requires=requirements["main"],
-      extras_require={"docs": requirements["docs"]},
+      install_requires=read_requirements('requirements/main.txt'),
```
Comment (TheMP) [inline setup.py:19]: @endrjuskr will this work without extras_require?

File: .github/workflows/workflow.yml (hunk 2 above)
Comment (radujica) [reply .github/workflows/workflow.yml:25]: I split the test requirements so the tests requirement also need to be cached (and installed)

File: setup.py (hunk 5 above)
Comment (radujica) [reply setup.py:19]: just pointing out that the github actions do work with this change

File: .github/workflows/workflow.yml [hunk 6]
```diff
@@ -54,14 +55,15 @@ jobs:
...
+        key: ${{ runner.os }}-pip-${{ hashFiles('requirements/main.txt', 'requirements/docs.txt', 'requirements/docs.txt') }}
         restore-keys: |
           ${{ runner.os }}-pip-
 
     - name: Install dependencies
       run: |
         pip install .
-        pip install -e .[docs]
+        pip install -r requirements/docs.txt
+        pip install -r requirements/tests.txt
```
Comment (radujica) [inline .github/workflows/workflow.yml:66]: docs might be also building docs for the tests; if that's the case, could exclude the tests and also not install the deps here

**Pull request merged**